### PR TITLE
Rack 2.0

### PR DESCRIPTION
**Pull request by natemontgomery**

### CODE REVIEW

**Comment by seerahulsingh** [comment]
@sshaw 
We were getting this error **cannot load such file -- rack/contrib/sendfile (LoadError)** 
 may be because of the line here -> https://github.com/sshaw/rack-contrib/blob/master/lib/rack/contrib.rb#L37, We removed this line and the file (sendfile.rb) both and now its working for us.

**Comment by sshaw** [comment]
Oh. Can't believe I missed that. Thanks. I've updated my branch/PR.

File: test/spec_rack_nested_params.rb [hunk 1]
```diff
@@ -37,10 +37,10 @@ def middleware
     req.put?.must_equal true
   end
 
-  specify "should make first boolean have precedence even after request already parsed" do
+  specify "should make last boolean have precedence even after request already parsed" do
     app = Rack::MethodOverride.new(middleware)
     req = app.call(form_post("foo=1&foo=0")).last
-    req.POST.must_equal({"foo" => '1'})
+    req.POST.must_equal({"foo" => "0"})
```
Comment (mojowen) [inline test/spec_rack_nested_params.rb:43]: this seems like a subtle and potentially dangerous change - why are changing our the test cases to accommodate?

**Review by mojowen — CHANGES_REQUESTED** [review]
Question about the test case change

File: test/spec_rack_nested_params.rb (hunk 1 above)
Comment (sshaw) [reply test/spec_rack_nested_params.rb:43]: This behavior will change with 2.0. See [this issue](https://github.com/rack/rack-contrib/issues/92) (and maybe [this too](https://github.com/rack/rack-contrib/pull/137)).  A deprecation warning was added to [v1.8.0](https://github.com/rack/rack-contrib/releases/tag/v1.8.0).

Comment (mojowen) [reply test/spec_rack_nested_params.rb:43]: @sshaw makes sense - thanks for the documentation

**Review by mojowen — APPROVED** [review]

**Pull request closed without merging**